File: trace_process/src/main/java/com/zhangyue/ireader/trace_1_2_3_7_process/MethodTrace.java
package com.zhangyue.ireader.trace_1_2_3_7_process;

import androidx.annotation.Keep;

import com.zhangyue.ireader.trace_1_2_3_7_process.handle.IMethodTraceHandle;
import com.zhangyue.ireader.trace_1_2_3_7_process.handle.MethodTraceHandle;

@Keep
public class MethodTrace {

    private static final IMethodTraceHandle METHOD_TRACE_HANDLE = new MethodTraceHandle();

    static {
        MethodTraceConfigKt.applyConfig();
    }

    public static void onMethodEnter() {
        METHOD_TRACE_HANDLE.onMethodEnter();
    }

    public static void onMethodExit(String str) {
        METHOD_TRACE_HANDLE.onMethodExit(str);
    }

}

File: trace_process/src/main/java/com/zhangyue/ireader/traceProcess/MethodTrace.kt
package com.zhangyue.ireader.traceProcess

import android.util.Log
import com.zhangyue.ireader.traceProcess.handle.IMethodTraceHandle
import com.zhangyue.ireader.traceProcess.handle.MethodTraceHandle
import com.zhangyue.ireader.traceProcess.handle.MethodTraceHandle.Companion.TAG

class MethodTrace {

    companion object {
        init {
            applyConfig()
            Log.i(
                TAG,
                "plugin config --> [onlyCheckMainThread1:$onlyCheckMainThread1 ," +
                        "infoConstThreshold1:$infoConstThreshold1," +
                        "warnConstThreshold1:$warnConstThreshold1," +
                        "errorConstThreshold1:$errorConstThreshold1]"
            )
        }

        /**
         * 当插件设置自定义处理类时，进行替换
         */
        @JvmStatic
        private val METHOD_TRACE_HANDLE: IMethodTraceHandle = MethodTraceHandle()

        /**
         * 方法入口
         */
        @JvmStatic
        fun onMethodEnter() {
            METHOD_TRACE_HANDLE.onMethodEnter()
        }


        /**
         * 方法出口
         */
        @JvmStatic
        fun onMethodExit(str: String) {
            METHOD_TRACE_HANDLE.onMethodExit(str)
        }
    }
}